### ISSUE TITLE
Allow users to delete email settings

### ISSUE THREAD

**Issue by senior** [metabase/metabase#7287]
As mentioned in https://github.com/metabase/metabase/issues/7213 we don't currently allow users to remove email settings. The form field validation will prevent it and the backend code implicitly requires it.

We should also users to remove email settings entirely via the UI.

**Comment by senior** [comment]
@salsakran @mazameli I looked into this and I think it might be more intuitive from the user perspective if we had a single click way of removing these settings, similar to how we treat caching. There are several different fields that need to be populated for email settings to work, which are currently marked as required. Today that is the SMTP host, port and from address. Not having a disable/clear/delete button/slider means we'll need to have some more subtle validation logic. If an smtp host is specified, then you must also specify the port and from address. If you don't specify a smtp host, do we keep the port and from address or clear it?

Seems like it would be cleaner to just have a separate path for that, separate endpoint that will clear it, then the logic can be, if it's enabled, these things are required, if not, clear it.

**Comment by kdoh** [comment]
Going to leave #5345 here just for additional context. I think I agree on the single removal action being preferable here.

**Comment by camsaul** [comment]
I don't think it makes sense to keep 3 issues (#5345, #7213, #7287) open for being able to delete email settings. Can we close out a couple of them?

**Comment by senior** [comment]
@camsaul I can close this in favor of #5435 as I opened this one not knowing we already had something for this. I do think #7213 is a separate issue as there was some problem updating settings, we're not sure what that problem is/was but only after they were unable to change the settings did they try to clear them. I was treating that as two separate issues, "Why can't I update my email settings?" and "It would be nice if I could just clear them".
